core/capabilities/compute/compute.go

```diff
@@ -116,9 +123,47 @@ func (c *Compute) Execute(ctx context.Context, request capabilities.CapabilityRe
 	return c.executeWithModule(m.module, config, request)
 }
 
-func (c *Compute) initModule(id string, binary []byte, workflowID, referenceID string) (*module, error) {
+func (c *Compute) initModule(id string, binary []byte, workflowID, workflowExecutionID, referenceID string) (*module, error) {
 	initStart := time.Now()
-	mod, err := host.NewModule(&host.ModuleConfig{Logger: c.log}, binary)
+	mod, err := host.NewModule(&host.ModuleConfig{
+		Logger: c.log,
+		Fetch: func(req *wasmpb.FetchRequest) (*wasmpb.FetchResponse, error) {
+			messageID, err := getMessageID(workflowID, workflowExecutionID)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get message ID: %w", err)
+			}
+
+			fields := req.Headers.GetFields()
+			headersReq := make(map[string]any, len(fields))
+			for k, v := range fields {
+				headersReq[k] = v
+			}
+
+			payloadBytes, err := json.Marshal(sdk.FetchRequest{
+				URL:       req.Url,
+				Method:    req.Method,
+				Headers:   headersReq,
+				Body:      req.Body,
+				TimeoutMs: req.TimeoutMs,
+			})
+			if err != nil {
+				return nil, fmt.Errorf("failed to marshal fetch request: %w", err)
+			}
+
+			resp, err := c.outgoingConnectorHandler.HandleSingleNodeRequest(context.Background(), messageID, payloadBytes)
```

Comment:
This should source a context linked to the lifecycle of the service at the very least. (Ideally it would source it from the request, but I think that's challenging given that wasmtime-go isn't context aware.

Reply:
true, I added this context.Background() to satisfy the method and then forgot to properly propagate it

Reply:
this will be addressed on: https://smartcontract-it.atlassian.net/browse/CAPPL-121